src/Framework/Traits.cs

```diff
@@ -425,5 +425,30 @@ public enum SdkReferencePropertyExpansionMode
             ExpandUnescape,
             ExpandLeaveEscaped
         }
+
+        /// <summary>
+        /// Emergency escape hatch. If a customer hits a bug in the shipped product causing an internal exception,
+        /// and fortuitously it happens that ignoring the VerifyThrow allows execution to continue in a reasonable way,
+        /// then we can give them this undocumented environment variable as an immediate workaround.
+        /// </summary>
+        /// <remarks>
+        /// Clone from ErrorUtilities which isn't (yet?) available in Framework.
+        /// </remarks>
+
+        private static readonly bool s_throwExceptions = String.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDDONOTTHROWINTERNAL"));
+
+        /// <summary>
+        /// Throws InternalErrorException.
+        /// </summary>
+        /// <remarks>
+        /// Clone of ErrorUtilities.ThrowInternalError which isn't (yet?) available in Framework.
+        /// </remarks>
+        internal static void ThrowInternalError(string message)
```

Comment:
Didn't you add this, admittedly with an inner exception and extra args?

Reply:
Hopefully over time we'll be able to reduce the duplication, but this is a more specific overload so I think it's ok to leave it for now.